smartsim/_core/_cli/build.py

```diff
@@ -285,73 +223,236 @@ def build_redis_ai(
         logger.info("ML Backends and RedisAI build complete!")
 
 
-def infer_torch_device() -> str:
+def infer_torch_device() -> _TDeviceStr:
     backend_torch_path = f"{CONFIG.lib_path}/backends/redisai_torch"
-    device = "cpu"
-    if Path(f"{backend_torch_path}/lib/libtorch_cuda.so").is_file():
-        device = "gpu"
-    return device
+    return (
+        "gpu" if Path(f"{backend_torch_path}/lib/libtorch_cuda.so").is_file() else "cpu"
+    )
 
 
-def install_torch(
-    build_env: BuildEnv, versions: Versioner, device: str = "cpu", verbose: bool = False
+def install_py_torch_version(
+    device: _TDeviceStr = "cpu",
+    fetch_if_missing: bool = False,
+    verbose: bool = False,
 ) -> None:
     """Torch shared libraries installed by pip are used in the build
     for SmartSim backends so we download them here.
     """
+    logger.info("Searching for a compatible TORCH install...")
+    if BuildEnv.is_macos():
+        end_point = None
+        device_suffix = ""
+    else:  # linux
+        end_point = "https://download.pytorch.org/whl/torch_stable.html"
+        if device == "cpu":
+            device_suffix = Versioner.TORCH_CPU_SUFFIX
+        elif device == "gpu":
+            device_suffix = Versioner.TORCH_CUDA_SUFFIX
+        else:
+            raise BuildError("Unrecognized device requested")
```

Comment:
will the linter complain here about an unnecessary else?

can we shift the validation to the start and short circuit out earlier?

```
if device not in ['cpu', 'gpu']:
    raise BuildError("Unrecognized device requested")
```